presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -551,10 +552,7 @@ public static Path createTemporaryPath(HdfsContext context, HdfsEnvironment hdfs
         // use a per-user temporary directory to avoid permission problems
         String temporaryPrefix = "/tmp/presto-" + context.getIdentity().getUser();
 
-        // use relative temporary directory on ViewFS
-        if (isViewFileSystem(context, hdfsEnvironment, targetPath)) {
-            temporaryPrefix = ".hive-staging";
-        }
+        checkState(!isViewFileSystem(context, hdfsEnvironment, targetPath), "Special support for viewfs has been removed, because it was erratic");
```

Comment:
Using `checkState` would result in an uncategorized error. Something like this would be better:
```java
throw new PrestoException(NOT_SUPPORTED, "Writing to locations on ViewFs is not supported: " + targetPath);